### ISSUE TITLE
Questions about learning rate setting and official evaluation script

### ISSUE THREAD

**Issue by saltyfsh** [zzaiyan/ECRformer#1]
Hi, thank you for releasing the ECRformer code.

I am trying to reproduce the results reported in the paper, but I have two questions about the official settings.

First, I noticed that the paper says the initial learning rate is `4e-4`, but in `train.py` the optimizer uses `lr * 5`. In this case, if `lr` is set to `4e-4`, the actual learning rate becomes `2e-3`. Could you please clarify why `lr` is multiplied by 5 in the code, and which learning rate was used for the reported results?

Also, the paper describes a learning-rate decay strategy based on validation loss, while the released code seems to use a different scheduler. Could you please clarify which scheduler was used to obtain the results in the paper?

Second, I could not find a complete official test script in the repository. Could you please provide the official test/evaluation script or command used to reproduce the quantitative results reported in the paper?

Thank you very much.

**Comment by zzaiyan** [comment]
Thank you for your interest and for pointing out the issues in the code! I will now address each point individually.

**Regarding the learning rate setting:** The current `*5` multiplier was added during the code cleanup and testing phase to quickly verify convergence. The learning rate we used to achieve the reported accuracy in the paper remains `4e-4`. You may simply remove the `*5` multiplier.

**Regarding the learning rate scheduler:** This was also introduced during the code cleanup phase to stabilize training; you may replace it with the code snippet below to align with the paper. Testing has shown that the learning rate scheduler has only a limited impact on the final results.

```python
def configure_optimizers(self):
    optimizer = torch.optim.AdamW(
        self.net.parameters(), lr=self.lr, weight_decay=1e-3)
    scheduler = torch.optim.lr_scheduler.ReduceLROnPlateau(
        optimizer, mode='min', factor=0.5, patience=10)  # The actual decay factor used is 0.5, which was erroneously listed as 0.1 in the paper.
    return {
        'optimizer': optimizer,
        'lr_scheduler': {
            'scheduler': scheduler,
            'monitor': 'valid_loss',
            'interval': 'epoch',
            'frequency': 1,
        },
    }
```

**Regarding the complete testing script:** We will release the evaluation and result export scripts in an upcoming update.

We are currently using the released version of the code (rather than the original code used during the submission phase) to reproduce the accuracy reported in the paper; this training and validation process will take some time to complete.

A version of the code with fixes for all the aforementioned issues—along with the complete testing scripts and pre-trained weights—will be updated and released shortly. Please stay tuned.
